### PR TITLE
Adds Minecarts

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Designed for use in #83163 , not implemented anywhere else currently. 

Mining cars (existing crate reskin) now have a hefty drag slowdown. 
Click-dragging them onto a minecart rail (new), however, will negate this slowdown, allowing you to drag them at nearly full speed. 
This also gives you the opportunity to give them a bump to push them down the tracks at a decent pace. 

https://github.com/tgstation/tgstation/assets/51863163/293adbfb-e2bd-49ba-beef-c998c388ecf0

Other notes: 
- Running a cable underneath a minecart (underfloor allowed) will power the rail, keeping its momentum, or boosting it slightly if it falls below certain thresholds. 
- If the minecart comes to an abrupt stop with a decent amount of speed, it will chuck all of its contents ahead a few tiles. 
- Break rails also exist, and stop the cart instantly (and with no contents chucking) when powered with cable.
- The cart does hurt people who are standing in the way if it's moving with moderate speed. 
- And it also hurts people lying on the tracks. 
- You can buckle yourself onto a (closed) minecart to ride it. Someone else has to give you a push though.

## Why It's Good For The Game

I was struggling to come up with a method of bulk transferring items for the linked PR that didn't seem too contrived (bluespace BS, pods), wouldn't require work refactoring an existing system (transit tubes), and wasn't a tram, and I realized we have mine carts but no rails. 

So I added rails and made the minecarts work on the rails. This gives a method of shipping items from point A to point B that fits well in certain settings, such as the lower areas of Icebox, or maybe in space ruins. 

I also think it would be fun to give miners some tools to construct rails, to let them ship boulders back to the station via railway rather than using teleportation. 

## Changelog

:cl: Melbert
add: Adds Minecarts, (possibly admin only depending on when this PR is merged in relation to the Icebox Bar PR) 
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
there should be a "launching rail" or something that allows people to click on it for the rail to give the cart a speed boost in a specific direction, just for rideable cart systems.

waiting for when these become buildable so i can make the worst station fast-travel system ever

**Comment by tralezab** [comment]
this is dope, can't wait to see what kind of horrible maint creation you're cooking

>waiting for when these become buildable

kinda surprised it's not buildable wow

**Comment by Sadboysuss** [comment]
Would be great if you could buckle people to the rails, finally the tram dilemma in ss13

**Comment by Wallemations** [comment]
> Minecarts w/ powered rails

 Craftbros, did we win?

**Comment by MrMelbert** [comment]
I forgot I was gonna add ramps before PRing this so it can go up Z-levels

**Comment by carlarctg** [comment]
one step closer to vents turning into the extraction mission from drg

**Comment by Admiral-Gigaton** [comment]
does hitting someone knock them back out of the way? and if so, can you make sure it launches them sideways? that way i cant just get someone on a rail, and infinitely do damage and kill them by them getting launched forwards, in the direction of the minecart (like how the tram throws your corpse typically sideways)

**Review by norsvenska — CHANGES_REQUESTED** [review]

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -150,10 +150,406 @@
...
+
+/obj/structure/minecart_rail/examine(mob/user)
+	. = ..()
+	. += rail_examine()
+
+/obj/structure/minecart_rail/proc/rail_examine()
+	return span_notice("Run a powered cable underneath it to power carts as they travel, maintaining their speed.")
+
+/obj/structure/minecart_rail/railbreak
+	name = "cart rail break"
```
[the remark this answers is not in the document]
Comment (Admiral-Gigaton) [reply code/modules/mining/mine_items.dm:546]: > should be 'brake', not 'break'

Minor spelling mistake

**Comment by scrunklershrungle932** [comment]
are rails or minecarts craftable ATM?

File: code/modules/mining/mine_items.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrMelbert) [reply code/modules/mining/mine_items.dm:546]: It's called break because it breaks the speed
```suggestion
	name = "cart rail brake"
```

**Comment by MrMelbert** [comment]
> does hitting someone knock them back out of the way? and if so, can you make sure it launches them sideways?

Already does

> are rails or minecarts craftable ATM?

No

**Pull request merged**